addons/Text_To_Speech/src/presenter.js

```diff
@@ -375,9 +375,56 @@ function AddonText_To_Speech_create() {
         presenter.speakWithCallback(texts, null);
     };
 
+    // Too long utterences may take much too long to load or exceed Speech Synthesis API character limit
+    presenter.splitLongTexts = function (texts) {
+        var newTexts = [];
+        var maxTextLen = 200;
+        for (var i = 0; i < texts.length; i++) {
+            if (texts[i].text !== null && texts[i].text !== undefined && texts[i].text.trim().length > 0) {
+                if (texts[i].text.trim().length > maxTextLen) {
+                    var sentences = texts[i].text.split(/[.,:;!?\/\\()]/);
+                    for (var j = 0; j < sentences.length; j++) {
+                        var sentenceLen = sentences[j].trim().length;
+                        if (sentenceLen > 0) {
+                            if (sentenceLen < maxTextLen) {
+                                newTexts.push({
+                                    text: sentences[j],
+                                    lang: texts[i].lang
+                                });
+                            } else {
+                                var maxSplitLen = sentenceLen / (Math.floor(sentenceLen/maxTextLen) + 1);
+                                var workString = '';
```

Comment:
To też mogłaby być podfunkcja, która zwraca tablicę słów o odpowiedniej długości.